code/datums/mind.dm

```diff
@@ -69,6 +69,8 @@
 
 	//fix scrying raging mages issue.
 	var/isScrying = 0
+	var/list/heard_before = list()
+	var/list/heard_before_names = list()
```

Comment:
Is this list necessary? You use it in the input(), but it should be possible to pick from the assoc list the same way.